src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs

```diff
@@ -181,10 +190,11 @@ internal static bool DeclaresTheObjectClass(this MetadataReader reader)
         }
 
         /// <exception cref="BadImageFormatException">An exception from metadata reader.</exception>
-        private static bool IsSystemObjectOrThrow(this MetadataReader reader, TypeDefinition typeDef)
+        internal static bool IsPublicTypeOrThrow(this MetadataReader reader, TypeDefinition typeDef, string namespaceName, string typeName)
```

Comment:
Should this just be "IsPublicType"?  I don't get the "OrThrow" bit (even in the old implementation).